### DISCUSSION TITLE
20220415

### DISCUSSION THREAD

**Discussion by tmiw** [ARSFI/paclink-open-source#18] [General]
Happy Tax Day!

This is a pre-production build intended to gauge current status of the project and determine what needs to be fixed. Brief testing indicates that packet (using Kenwood TM-D700A) and Telnet channels work; status of other channel types is TBD.

Contains (vs. [last pre-release version](https://github.com/ARSFI/paclink-open-source/releases/tag/nightly_20220327)):

* Resolve issue with AGW modem preventing proper dequeuing of already sent packets (and thus preventing gateway login).
* Various cleanup of "unresolved exception" errors in TCP/IP handling.

Instructions:

1. Unzip paclink-open-source.zip somewhere on your hard drive.
2. Run setup.exe to install on your system. Paclink will be installed in the Start menu.
3. Open Paclink from the Start menu and reconfigure if required. (There is no import functionality from the existing Paclink at this time but if you used a pre-release version, configuration should be saved.)

Please leave comments in the release's discussion thread or email K6AQ@winlink.org with feedback. 👍 

<hr /><em>This discussion was created from the release <a href='https://github.com/ARSFI/paclink-open-source/releases/tag/nightly20220415'>20220415</a>.</em>

**Comment by kd4vrf** [comment]
I was able to install this release and get it working with Telnet. I also tried using my Pactor modem but it failed connecting to the Pactor Modem with the following:

2022-05-12 14:48:39.2566 | DEBUG | Main | *** Initializing the PTC DR-7800 TNC
2022-05-12 14:48:39.7990 | ERROR | SCSHostPort | [SCSHostPort.Startup] Empty path name is not legal. (Parameter 'path')
2022-05-12 14:48:40.2526 | DEBUG | Main | *** PTC Fault
2022-05-12 14:48:40.2667 | DEBUG | Main | *** Error during PTC DR-7800 startup... See error log
2022-05-12 14:48:40.2667 | DEBUG | Main | *** TNC PTC DR-7800 on serial port COM7 initialization failed
2022-05-12 14:48:40.2888 | DEBUG | Main | *** Closing Pactor - KC0TPS at 2022/05/12 18:48:39
2022-05-12 14:48:44.4369 | DEBUG | Main | *** Closing Pactor - KC0TPS at 2022/05/12 18:48:43

I'm not sure if this is expected to work in this build. If it is let me know and I will keep trying.

When you click "help --> Contents" nothing loads, I just get a spinning indicator like it's about to load and then nothing.

For basic Telnet functionality I didn't see any issues. Ran it for about 4 hours and sent myself a few Winlink messages and it worked as expected.

OS: Windows 10 (VM on a Mac)
Mail Client: Windows Mail Version 16005.14326.20858.0

**Reply by kd4vrf** [reply]
I was able to find the ".aps" TNC Setup file in the directory structure. Once I specified the ".aps" file for the DR-7800 I was able to connect to the Pactor modem. I wasn't able to fully connect to an RMS gateway because of the FSK/PSK Levels and the inability to set Long Path. See below for details on both issues. I was able to resolve the FSK/PSK levels by editing the ".aps" TNC configuration file, but I couldn't resolve the Long Path issue.

Long Path:
I use a Flex SDR Radio and the latency is higher than other non SDR radios. I tried messing with the CSDelay and TXDelay but I wasn't able to connect in either Winlink Express or PacLink. In Winlink Express there is an option to set Long Path mode, basically it puts a "!" in front of the call sign you are connecting to so the Pactor modem knows to initiate the connection in Long Path mode. I am only able to make connections to an RMS gateway with Long Path set regardless of the CSDelay or TXDelay.

In Winlink Express the Long Path option simply adds a "!" to the call sign when you connect. I did try to just add the "!" manually to the call sign in the GUI for PacLink but there is a check for valid call signs that won't allow the "!". For example when trying to connect to RMS Gateway "!KC0TPS" I get the following error:

"!KC0TPS is not a valid amateur or MARS callsign..."

If you add an option to set Long Path mode, or you allow the "!" to be before the remote callsign in the existing GUI, SDR radios that have a higher latency would be able to connect using Pactor.


FSK/PSK Levels:
For my Pactor DR-7800 I need to set the FSK/PSK levels to higher than the max values that the PacLink GUI will allow. I was not able to generate more than 5 watts of transmit power due to the low audio levels when using the GUI. In Winlink Express I use 5000mV. I was able to set those values in PacLink using the ".aps" TNC setup file using the following values:

FSKA 5000 ; Not required...set in Pactor Channel menu for all PTC Models WM4J Edits
PSKA 5000 ; WM4J Edits 

With the above values set in the ".aps" TNC setup file I was able to generate the proper audio levels and generate full power from my radio.

The Max you can set the FSK/PSK levels to in the PacLink GUI is:

FSK = 1000
PSK = 1009

In Winlink Express the Max values are:

FSK = 9000
PSK = 9000

If you can increase the default values in the GUI to match Winlink Express and the capabilities of the newer Pactor Modems you wouldn't have to edit the ".aps" file to set the levels. Either way setting them through the ".aps" TNC setup file worked to override the GUI settings and allowed me to generate the proper power from my radio.

**Comment by tmiw** [comment]
Hi @kd4vrf, thanks so much for the clarification! I went ahead and did the following:

1. Increased the maximum for the FSK and PSK options in the Pactor Channels window to 9000, and
2. Added a checkbox in the UI for Long Path. Example below:

<img width="798" alt="Screen Shot 2022-05-15 at 8 07 01 PM" src="https://user-images.githubusercontent.com/449242/168513527-d3f7c202-e3ee-4d5a-b00f-c0ea886beb88.png">

Can you try the release I just generated (https://github.com/ARSFI/paclink-open-source/releases/tag/nightly_20220515) and let me know if that works for you?

**Reply by kd4vrf** [reply]
I changed my ".aps" file back to the default DR_7800 file so I wouldn't override the FSK/PSK levels in the GUI.

I selected both the "Long Path" check box and I adjusted the FSK/PSK values to 5000mV.

I connected successfully but I didn't have any queued messages to download. I went ahead and sent myself a message and tried to connect a second time.

The second connection initiated but half way into downloading the message PacLink crashed. See screen shot below:

![Screen Shot 2022-05-16 at 12 34 41 AM](https://user-images.githubusercontent.com/45251519/168521510-b1026ee4-8b2f-4134-af6c-832c2f36047c.png)

I rebooted my windows VM and Pactor modem. The third try I was able to connect and download the message successfully:

![Screen Shot 2022-05-16 at 12 40 29 AM](https://user-images.githubusercontent.com/45251519/168521677-27534058-168f-4dd4-8c1f-e5523f45aa7c.png)

I will try it a few more times tomorrow morning and see if the crash occurs again.

**Reply by tmiw** [reply]
> I will try it a few more times tomorrow morning and see if the crash occurs again.

Sounds good, let me know!

**Reply by tmiw** [reply]
BTW if you can send over the logs that would be good. They are in C:\Users\\[your username]\AppData\Roaming\Paclink under the Logs and ErrorLogs folders. The paclink.db file is there too in case you ever need to wipe your configuration and start from scratch.

**Reply by kd4vrf** [reply]
I responded over on the 20220515 discussion thread to keep things organized. I attached the logs over on that discussion.
